rust/arrow/src/ipc/gen/mod.rs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#![allow(non_snake_case)]
+
+pub mod File;
```

Comment:
Do the 3 issues stop it from compiling altogether? When I worked on feather support eariler in the year, I didn't have any compilation issues (unless it's issues in the sense of CI being strict), but I had to still work around the nested modules.

Do you mind sharing the `flatc` script/command that you used when building? We should probably document that somewhere.

I think in the near future, we might need to add a `build.rs` to auto-generate fbs and proto files, I can do that as I want to add support for a Flight client in the next release.

Reply:
I just pushed a commit to add instructions to the README so you can see the steps I took.

Reply:
Thanks, I'll take a look later today

Reply:
I ended up taking longer, my apologies